server/signal_test.go

```diff
@@ -59,3 +61,226 @@ func TestSignalToReOpenLogFile(t *testing.T) {
 		t.Fatalf("Expected log to contain %q, got %q", expectedStr, string(buf))
 	}
 }
+
+func TestSignalToReloadConfig(t *testing.T) {
+	opts, err := ProcessConfigFile("./configs/reload/basic.conf")
+	if err != nil {
+		t.Fatalf("Error processing config file: %v", err)
+	}
+	s := RunServer(opts)
+	defer s.Shutdown()
+
+	loaded := s.ConfigTime()
+
+	// This should cause config to be reloaded.
+	syscall.Kill(syscall.Getpid(), syscall.SIGHUP)
```

Comment:
To avoid flappers, should you not wait before the reload to make sure that time before and after will be different?